### ISSUE TITLE
PLE1310 fails on `bytes.strip`

### ISSUE THREAD

**Issue by dscorbett** [astral-sh/ruff#15968]
### Description

[`bad-str-strip-call` (PLE1310)](https://docs.astral.sh/ruff/rules/bad-str-strip-call/) in Ruff 0.9.4 flags a `bytes.strip` call when its argument is a `str`, but that is a false positive because `bytes.strip` does not accept a `str` argument.
```console
$ cat >ple1310.py <<'# EOF'
b"".strip("//")
# EOF

$ python ple1310.py 2>&1 | tail -n 1
TypeError: a bytes-like object is required, not 'str'

$ ruff --isolated check --select PLE1310 ple1310.py --output-format concise
ple1310.py:1:11: PLE1310 String `strip` call contains duplicate characters
Found 1 error.
```
PLE1310 does not flag a `bytes.strip` call when its argument is a `bytes`, which a false negative.
```console
$ echo 'b"".strip(b"//")' | ruff --isolated check --select PLE1310 -
All checks passed!
```
